### ISSUE TITLE
Feature request: platform hints

### ISSUE THREAD

**Issue by modulovalue** [dart-lang/pub#2130]
Flutter can run on many platforms. 
To successfully utilize that, one still has to inject platform-specific dependencies because packages that claim to support flutter do not necessarily support every platform that Flutter can run on.

Now, with the technical preview of [flutter_web](flutter.dev/web) being released and googles _'promise'_ to support [flutter_desktop](https://flutter.dev/desktop) I foresee that it will become increasingly more difficult to discover packages for each platform.

For example, these two packages [launcher_assist](https://pub.dev/packages/launcher_assist) &  [android_alarm_manager](https://pub.dev/packages/android_alarm_manager) are not supported on iOS. Developers shouldn't be forced to rely on the SEO skills of a package creator to be able to find packages for each platform.

Instead of waiting for packages like [flutter_web_view](https://pub.dev/packages/flutter_web_view) to support Linux/Windows/MacOS/Web (which will obviously not be an easy task and quite unreasonable to expect) I believe the community would be better off if they could easily discover and pick platform-specific packages to inject into their flutter core.

### What should a solution look like?

A naive but maybe _good enough_ solution would be to add support for canonical platform tags to packages that could be used to help discovery on pub.dev.

Example:
```yaml
platform-hint
  - ios
  - macos
```
But I'm unsure if the differences between different Linux distributions and browsers (or different versions of the platform that a package is supposed to run on) could make this a bad solution.

Anyways, If there's not already a discussion around this I'd like to spark one now. 

### TLDR
pub.dev should be able to allow discovery of platform-specific (flutter) packages.
